### PR TITLE
[bugfix]: Inform about different version

### PR DESCRIPTION
**Pull request by GermanBluefox**
### For Bugfixes:
Sometimes the update of adapter happens without "upload" command. To detect that versions in DB and on disk differs, this bugfix writes a warning int log in this case.

Additionally it helps split this PR https://github.com/ioBroker/ioBroker.js-controller/pull/2875 into really essential changes

### CODE REVIEW

File: packages/adapter/src/lib/adapter/adapter.ts [hunk 1]
```diff
@@ -11519,6 +11519,14 @@ export class AdapterClass extends EventEmitter {
 
         this.adapterConfig = adapterConfig;
 
+        // Check that version in DB is the same as on disk
+        if ((adapterConfig as ioBroker.InstanceObject).common.version !== packJson.version) {
+            // TODO: think about to make upload automatically if a version on disk is newer than in DB. Now it is just hint in the log.
+            this._logger.warn(
+                `${this.namespaceLog} Version in DB is ${(adapterConfig as ioBroker.InstanceObject).common.version}, but this version is ${packJson.version}. Please synchronise the adapter with "iob upload ${(adapterConfig as ioBroker.InstanceObject).common.name}".`
```
Comment (Apollon77) [inline packages/adapter/src/lib/adapter/adapter.ts:11526]:
```suggestion
                `${this.namespaceLog} Version in DB is ${(adapterConfig as ioBroker.InstanceObject).common.version}, but installed version is ${packJson.version}. Please synchronise the adapter with "iob upload ${(adapterConfig as ioBroker.InstanceObject).common.name}".`
```

**Review by Apollon77 — APPROVED** [review]
I would just adjust the wording a bit

**Comment by foxriver76** [comment]
As far as I know the case where this happens is GitHub installs where we currently are aware of and did this on purpose until now. So this will always lead to this scenario if we don’t want this anymore it would be more like should we also do the upload on GH installs? (URL command)

**Comment by Apollon77** [comment]
Also github installs shoukld have an auto upload ...in my eyes there should be no case anymore where it is not executed

**Comment by foxriver76** [comment]
> Also github installs shoukld have an auto upload ...in my eyes there should be no case anymore where it is not executed

Then I would tend to say let’s fix this instead of adding a log warning where we know why it happens. Or are there other cases where we are sure that it was no url install (Admin npm or GitHub dialog)

**Comment by Apollon77** [comment]
maybe the log helps to see if there are cases any ore :-) as said ... should not

**Comment by GermanBluefox** [comment]
This is targeting to help to find the error with installations from git/npm so the upload is not executed automatically after installation

**Comment by foxriver76** [comment]
is already integrated with the auto upload + log

**Pull request closed without merging**